config/resolve_credentials_test.go

```diff
@@ -577,6 +577,8 @@ func TestResolveCredentialsIMDSClient(t *testing.T) {
 			opts := []func(*LoadOptions) error{
 				WithRetryer(func() aws.Retryer { return aws.NopRetryer{} }),
 				WithHTTPClient(httpClient),
+				WithSharedConfigProfile("profile-name"),
+				WithSharedConfigFiles([]string{"creds-file"}),
```

Comment:
Let's be more explicit about shutting this off - specifically, by giving it a profile name that categorically cannot exist, and no config files.

```suggestion
                                // separate from the local config, should it exist - the default loader will ignore nonexistent profiles
				WithSharedConfigProfile(" "),
				WithSharedConfigFiles([]string{}),
```